cli/cppcheckexecutor.cpp

```diff
@@ -199,23 +198,20 @@ int CppCheckExecutor::check(int argc, const char* const argv[])
 {
     CheckUnusedFunctions::clear();
 
-    CppCheck cppCheck(*this, true, executeCommand);
-
-    const Settings& settings = cppCheck.settings();
-    mSettings = &settings;
-
-    if (!parseFromArgs(&cppCheck, argc, argv)) {
-        mSettings = nullptr;
+    Settings settings;
+    if (!parseFromArgs(settings, argc, argv)) {
         return EXIT_FAILURE;
     }
     if (Settings::terminated()) {
-        mSettings = nullptr;
         return EXIT_SUCCESS;
     }
 
-    int ret;
+    CppCheck cppCheck(*this, true, executeCommand);
+    cppCheck.settings() = settings;
+    mSettings = &settings;
```

Comment:
It does not feel very robust that we set mSettings to a local variable like this.. but let's not refactor that now..

Reply:
Yeah, we should be be using dedicated `ErrorLogger` objects instead of making the class a multi-purpose one. Haven't looked into it yet.